src/api/app/models/user.rb

```diff
@@ -675,6 +675,46 @@ def owned_packages
     owned
   end
 
+  def owned_item(owner, filters = {})
+    if !owner.package.nil?
+      owner.package if filters.key?(:involved_packages)
+    elsif filters.key?(:involved_projects)
+      owner.project
+    end
+  end
+
+  def involved_items_as_owner(filters = {})
+    items = []
+
+    OwnerSearch::Owned.new.for(self).each do |owner|
+      new_item = owned_item(owner, filters)
+      items << new_item unless new_item.nil? || (filters[:search_text].present? && !new_item.to_s.match?(Regexp.escape(filters[:search_text])))
+    end
```

Comment:
This iteration can be split into several small steps:

We can get the `new_items` with a `map`:
```ruby
OwnerSearch::Owned.new.for(self).map{ |o| owned_item(o, filters) }
```

Then we can `reject` nils with:
```ruby
.reject(&:nil?)
```

and then we can reject the second part of the condition with:

```ruby
.reject { |new_item| filters[:search_text].present? && !new_item.to_s.match?(Regexp.escape(filters[:search_text])) }
```

If I'm not wrong, if we don't have a `:search_text` filter, the second part of the condition will never happen, so we could skip this last step if we have no `filters[:search_text]`:

```ruby
items = OwnerSearch::Owned.new.for(self)
  .map{ |o| owned_item(o, filters) } # Transform to owned items
  .reject(&:nil?)
if filters[:search_text].present?
  # Filter by the search terms only if we have the filters set
  items.reject! { |new_item| !new_item.to_s.match?(Regexp.escape(filters[:search_text])) }
end
items 
```

Reply:
I don't know if this snippet works, but looks like right and definitely more readable.

Reply:
That helped too. I used `map` as you proposed. I also moved the condittion of the `search_text` filter into the `owned_item` method. This prevents looping twice over the items. Please have a look again.